src/Timer_Obscure_Behaviour.md

```diff
@@ -32,99 +32,63 @@ On **CGB**:
-        END IF
-    END IF
-```
-
-The sentence marked with a (\*) has a different behaviour in GBC (AGB
-and AGS seem to have strange behaviour even in the other statements).
-When enabling the timer and maintaining the same frequency it doesn't
-glitch. When disabling the timer it doesn't glitch either. When another
-change of value happens (so timer is enabled after the write), the
-behaviour depends on a race condition, so it cannot be predicted for
-every device.
-
-## Timer Overflow Behaviour
-
-When TIMA overflows, the value from TMA is loaded and IF timer flag is
-set to 1, but this doesn't happen immediately. Timer interrupt is
-delayed 1 M-cycle from the TIMA overflow. The TMA reload to
-TIMA is also delayed. For 1 M-cycle, after overflowing TIMA, the value
-in TIMA is $00, not TMA. This happens only when an overflow happens, not
-when the upper bit goes from 1 to 0, it can't be done manually writing
-to TIMA, the timer has to increment itself.
-
-For example (SYS here is the lower 8 bits of the system counter):
-
-    Timer overflows:
-
-                  [A] [B]
-    SYS  FD FE FF |00| 01 02 03
-    TIMA FF FF FF |00| 23 23 23
-    TMA  23 23 23 |23| 23 23 23
-    IF   E0 E0 E0 |E0| E4 E4 E4
-
-    Timer doesn't overflow:
-
-                  [C]
-    SYS  FD FE FF 00 01 02 03
-    TIMA 45 45 45 46 46 46 46
-    TMA  23 23 23 23 23 23 23
-    IF   E0 E0 E0 E0 E0 E0 E0
-
-- During the strange cycle \[A\] you can prevent the IF flag from being
-set and prevent the TIMA from being reloaded from TMA by writing a value
-to TIMA. That new value will be the one that stays in the TIMA register
-after the instruction. Writing to DIV, TAC or other registers won't
-prevent the IF flag from being set or TIMA from being reloaded.
-
-- If you write to TIMA during the M-cycle that TMA is being loaded to it
-\[B\], the write will be ignored and TMA value will be written to TIMA
-instead.
-
-- If TMA is written the same M-cycle it is loaded to TIMA \[B\], TIMA is
-also loaded with that value.
-
-- This is a guessed schematic to explain the priorities with registers
-TIMA and TMA:
-
-![](imgs/timer_tima_tma_detailed.svg "imgs/timer_tima_tma_detailed.svg")
-
-TMA is a latch. As soon as it is written, the output shows that value.
-That explains that when TMA is written and TIMA is being incremented,
-the value written to TMA is also written to TIMA. It doesn't affect the
-IF flag though.
+Notice how the bits themselves are connected to the multiplexer and then to the falling-edge detector; this causes a few odd behaviors:
+
+- Resetting the entire system counter (by writing to `DIV`) can reset the bit currently selected by the multiplexer, thus sending a "Timer tick" and/or "[DIV-APU event](<#DIV-APU>)" pulse early.
+- Changing which bit of the system counter is selected (by changing the "Clock select" bits of [`TAC`]) from a bit currently set to another currently reset, will send a "Timer tick" pulse.
```

Comment:
"from a bit currently set to another currently reset" sounds a bit strange to me.. what is a "bit currently reset"? Could we reword?

Reply:
"A bit which is currently set" and "a bit which is currently reset"? Or "a bit equal to 1" and "a bit equal to 0"?

What matters here is the values contained in the bits that the multiplexer is considering. (Two, because the muxer is switching from one to the other.)

Reply:
Then I would use `unset`, or `currently set to 0`. A bit `currently reset` is confusing to read (as you would normally say `to reset a bit`, but less often a `reset bit` instead of a `bit set to 0`) and seems also to imply an action the preceded this situation (someone reset it), which may be true, but it's simply not useful as an information. If a bit is 0, something re-et it at some point, yes.

Reply:
```suggestion
- Changing which bit of the system counter is selected (by changing the "Clock select" bits of [`TAC`]) from a bit currently set to another that is currently unset, will send a "Timer tick" pulse.
```